src/main/java/algorithms/curated170/medium/meetingscheduler/MeetingScheduler.java

```diff
@@ -0,0 +1,40 @@
+package algorithms.curated170.medium.meetingscheduler;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+class MeetingScheduler {
+    List<Integer> minAvailableDuration(int[][] p1, int[][] p2, int duration) {
+
+        Arrays.sort(p1, ((a, b) -> a[0] - b[0]));
+        Arrays.sort(p2, ((a, b) -> a[0] - b[0]));
+
+        int i1 = 0, i2 = 0;
+        while (i1 < p1.length && i2 < p2.length) {
+            int s1 = p2[i1][0], e1 = p2[i1][1], s2 = p2[i2][0], e2 = p2[i2][1]; 
+
+            if (s2 >= e1) {
+                i1++;
+            } else if (s1 >= e2) {
+                i2++;
+                continue;
+            }
```

Comment:
I think you add this to prevent unnecessary max-min calculation.
 As it does not affect the efficiency, it is up to you removing or not.

Reply:
![image](https://user-images.githubusercontent.com/60903744/118374395-8be21880-b5c4-11eb-82ac-2fb90f19b008.png)

Reply:
The unnecessary calculations done otherwise aren't that worse than calculating these. I will remove these then, for the sake of briefness.

Reply:
> The unnecessary calculations done otherwise aren't that worse than calculating these. I will remove these then, for the sake of briefness.

You choose the right thing to do as expected. 😊